internal/agentdeployer/factory.go

```diff
@@ -111,43 +115,25 @@ func selectAgentDeployerType(options FactoryOptions) (string, error) {
 	return agentDeployerName, nil
 }
 
-// FindDevDeployPath function returns a path reference to the "_dev/deploy" directory.
-func FindDevDeployPath(options FactoryOptions) (string, error) {
-	dataStreamDevDeployPath := filepath.Join(options.DataStreamRootPath, options.DevDeployDir)
-	info, err := os.Stat(dataStreamDevDeployPath)
-	if err == nil && info.IsDir() {
-		return dataStreamDevDeployPath, nil
-	} else if err != nil && !errors.Is(err, os.ErrNotExist) {
-		return "", fmt.Errorf("stat failed for data stream (path: %s): %w", dataStreamDevDeployPath, err)
-	}
-
-	packageDevDeployPath := filepath.Join(options.PackageRootPath, options.DevDeployDir)
-	info, err = os.Stat(packageDevDeployPath)
-	if err == nil && info.IsDir() {
-		return packageDevDeployPath, nil
-	} else if err != nil && !errors.Is(err, os.ErrNotExist) {
-		return "", fmt.Errorf("stat failed for package (path: %s): %w", packageDevDeployPath, err)
-	}
-
-	return "", fmt.Errorf("\"%s\" %w", options.DevDeployDir, os.ErrNotExist)
-}
-
-func findAgentDeployers(devDeployPath string) ([]string, error) {
-	fis, err := os.ReadDir(devDeployPath)
+func findAgentDeployers(devDeployPath, expectedDeployer string) ([]string, error) {
```

Comment:
Nit. This method returns now a single deployer or none, right? We could change it to return a single value, and we could return a named error in case of no deployer found.
```suggestion
func findAgentDeployer(devDeployPath, expectedDeployer string) (string, error) {
```